### PR TITLE
providers: add okta

### PR DESCRIPTION
**Pull request by abraithwaite**

### CODE REVIEW

File: oauthproxy.go [hunk 1]
```diff
@@ -482,7 +482,11 @@ func (p *OAuthProxy) SignIn(rw http.ResponseWriter, req *http.Request) {
 		p.SaveSession(rw, req, session)
 		http.Redirect(rw, req, redirect, 302)
 	} else {
-		p.SignInPage(rw, req, 200)
+		if p.SkipProviderButton {
+			p.OAuthStart(rw, req)
+		} else {
+			p.SignInPage(rw, req, 200)
+		}
```
Comment (abraithwaite) [inline oauthproxy.go]: I think this is what #284 is referring to.

**Comment by abraithwaite** [comment]
Could still use some tests.  Lemme know if you have any specific rules about contributing and I'm happy to comply.

File: oauthproxy.go (hunk 1 above)
Comment (ploxiln) [inline oauthproxy.go]: It would be better for this to be in a separate PR, not mixed in with this one.

**Comment by Fodoj** [comment]
This is very much needed!

**Comment by asms** [comment]
@abraithwaite Nice!

**Comment by statik** [comment]
See issue #428

**Comment by asms** [comment]
@abraithwaite Validation does not work. Specifically request to `/oauth2/v1/userinfo` returns a 400 Bad Request. The reason is that the access_token is passed as a GET argument rather than using the `Authorization: Bearer ...` header. At a glance, it seems like `ValidateSessionState ` needs to be implemented (see https://github.com/bitly/oauth2_proxy/blob/a9c55bd6d1caeaa7ba20236b2e85129c8219b865/providers/facebook.go#L78 for an example).

**Comment by abraithwaite** [comment]
Thanks.  I'll try and look at it when I have time, otherwise if someone else beats me to it please close this PR!

**Comment by rothgar** [comment]
Commenting here to show support for this and follow updates

**Comment by rothgar** [comment]
I got this built but having trouble setting up the okta side of the application for testing. I think I'm not setting up my login redirect URI properly but if someone has an example it'd be helpful.
My requests are redirecting properly but I'm getting this on Okta
image

I see the request redirect to `GET - "/oauth2/start?rd=%2F" HTTP/1.0` but I still receive a 400 from okta.

**Comment by reedloden** [comment]
Since I can't edit this PR directly, opened #528.

**Comment by abraithwaite** [comment]
Neat!  Been waiting to close this.  Thanks!
